src/app/services/auth/auth.service.ts

```diff
@@ -0,0 +1,46 @@
+import {Injectable} from '@angular/core';
+import {environment} from '../../../environments/environment.prod';
+import {HttpClient, HttpErrorResponse} from '@angular/common/http';
+import {User} from '../../models/user';
+import {throwError} from 'rxjs';
+import {catchError, tap} from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthService {
+
+  constructor(private httpClient: HttpClient) { }
+
+  private logInUrl = '/login';
+  baseUrl = environment.baseUrl;
+
+  public logIn(user: User) {
+    return this.httpClient.post(this.baseUrl + this.logInUrl, user, {observe: 'response'})
+      .pipe(
+        tap(data => {
+          sessionStorage.setItem('authorization', data.headers.get('Authorization'));
+          console.log(sessionStorage.getItem('authorization'));
+        }),
+        catchError((err) => throwError(err))
```

Comment:
Is there any point in catching  error if you rethrow it again?

Reply:
Fixed